src/codegen.cpp

```diff
@@ -3839,9 +3851,17 @@ static bool emit_builtin_call(jl_codectx_t &ctx, jl_cgval_t *ret, jl_value_t *f,
                     *ret = emit_getfield_knownidx(ctx, obj, 0, utt, order);
                     return true;
                 }
+                else {
+                    Value *index = ctx.builder.CreateCall(prepare_call(jlfieldindex_func),
+                          {emit_typeof(ctx, obj, false, false), boxed(ctx, fld), ConstantInt::get(getInt32Ty(ctx.builder.getContext()), 0)});
```

Comment:
Does LLVM then unbox the fld value, before this actually reaches real code? This PR is meant to remove an allocation here right?

Can you also leave a comment explaining that?

Reply:
Yeah I think anything where the _reason_ for the core isn't obvious or straightforward should have a comment. 👍 Thanks